paddle/fluid/framework/operator.cc

```diff
@@ -1099,6 +1102,19 @@ Scope* OperatorWithKernel::PrepareData(
       auto* trans_var = new_scope->Var(var_name);
       input_vars[i] = trans_var;
 
+      if (is_inplace) {
+        for (auto& out_var_name_item : Outputs()) {
+          std::vector<Variable*>& output_vars =
+              ctx->outputs[out_var_name_item.first];
+          for (size_t i = 0; i < out_var_name_item.second.size(); ++i) {
+            auto& out_var_name = out_var_name_item.second[i];
+            if (out_var_name == var_name) {
+              output_vars[i] = input_vars[i];
```

Comment:
here output_vars[i] = input_vars[i] is incorrect. "i" in the index for output_vars[] and should NOT apply to input_vars[] which is totally independant with output_vars[]. Actually this local var "i" in used in both two nested "for" which is not a GOOD practice.

Reply:
@jianhang-liu Thanks for you reminder, i will use "output_vars[i] = trans_var;" here, due to line 1103 "input_vars[i] = trans_var", which will be safer and make sense.

Reply:
@xiaolil1 output_vars[i] = input_vars[i] is NOT correct, not only "unsafe". Unfortunately there are two "i" defined in nested for loop. [The first "i"](https://github.com/PaddlePaddle/Paddle/blob/8d0d6185ce3b3c7571b34316152ece5dd361f859/paddle/fluid/framework/operator.cc#L1049) is index for input_vars[]  and [the second "i"](https://github.com/PaddlePaddle/Paddle/blob/8d0d6185ce3b3c7571b34316152ece5dd361f859/paddle/fluid/framework/operator.cc#L1109) (in inner loop) is index for output_vars[].
So the code "[output_vars[i] = input_vars[i]](https://github.com/PaddlePaddle/Paddle/blob/8d0d6185ce3b3c7571b34316152ece5dd361f859/paddle/fluid/framework/operator.cc#L1112)" doesn't make sense.
As mentioned in my last comment, duplicated "i" in nested loop is NOT a good practice.